### ISSUE TITLE
PHP Warning in Custom Fields Sync

### ISSUE THREAD

**Issue by SirLouen** [polylang/polylang#378]
## Prerequisites

- [x] This is not a usage question (Those should be directed to the [community supported forum](https://wordpress.org/support/plugin/polylang), unless this is a question about Polylang Pro in which case you should [use the helpdesk](https://polylang.pro/support/)).
- [x] I have searched for similar issues in both open and closed tickets and cannot find a duplicate.
- [x] The issue still exists against the latest `master` branch of Polylang and the latest WordPress version.
- [ ] This bug happens with only Polylang plugin active
- [x] This bug happens with a default WordPress theme active
- [ ] I can reproduce this bug consistently

## Steps to reproduce the issue

1. Activate Sync for Custom Fields
2. Access certain post in my site

## Expected behavior and actual behavior

When I activate the Sync for Custom Fields option and go to certain post in my site, suddenly it appears this error messages:

```
PHP Warning:  array_keys() expects parameter 1 to be array, boolean given in /var/www/site/wp-content/plugins/polylang/modules/sync/sync-post-metas.php on line 46
PHP Warning:  array_merge(): Argument #2 is not an array in /var/www/site/wp-content/plugins/polylang/modules/sync/sync-post-metas.php on line 46
PHP Warning:  array_unique() expects parameter 1 to be array, null given in /var/www/site/wp-content/plugins/polylang/modules/sync/sync-post-metas.php on line 46
PHP Warning:  Invalid argument supplied for foreach() in /var/www/site/wp-content/plugins/polylang/modules/sync/sync-post-metas.php on line 46
```

The line is:

` foreach ( $keys = array_unique( array_merge( array_keys( $metas ), array_keys( get_post_custom( $to ) ) ) ) as $k => $meta_key ) {`

After some debug I've found that the issue is here:
`get_post_custom( $to )`
That is returning "false" in the post page, and an array in the action=edit page.

The problem is that I cannot reproduce this bug consistently in other posts. In fact I've created the same exact post again by copying the HTML content, title, categories and all that, and it doesn't show that error.

In fact, I think I could just delete the post and everything will be fine (or not) in the future.

But maybe this could serve for debugging purposes? Not sure.

The fact is that I'm not sure how this post has been "tagged" to be synced. It has been untouched for ages and suddenly today I found this.

If anyone would like to offer me an idea on how I could track down this issue, I could try to help on this.

**Comment by SirLouen** [comment]
Found the bug
I migrated from a WPML installation and maybe there are some remainders
Checked the variable **tr_ids** in modules/sync/sync_metas.php in line 206 (pre-2.6 function update_meta). And here is the data FYI:

```
 array(2) {
 ["en"]=>int(2632)
 ["es"]=> int(0)
}
```
This said, maybe it could be adequate to filter out "0" ids before launching the "get_metas_to_copy" function or during the launch of it.

Meanwhile I have to discover where is this stored to readapt it. If anyone can give me the exact position in the database would save me a lot of time of research :) 
I've checked in the wp_postmeta but can't seem to find it.
